test/e2e/helpers_test.go

```diff
@@ -387,6 +388,19 @@ func verifyPostgres(ctx context.Context, t *testing.T, env environments.Environm
 	require.GreaterOrEqual(t, migrationJob.Status.Succeeded, int32(1))
 }
 
+// containConfigPushCountMetric check if the prometheus metrics contains metrics for pushing configurations,
+// which indicate that this instance is the leader.
+func containConfigPushCountMetric(t *testing.T, url string, client *http.Client) bool {
+	req, err := http.NewRequest("GET", url, nil)
+	require.NoError(t, err)
```

Comment:
I brought this up already elsewhere but let me reiterate on this: I'd try not to use `assert` or `require` calls in functions like `require.Eventually` (in which context  `containConfigPushCountMetric` is called in).

The reason for this is that the whole test will be marked as failed regardless of the final result of `.Eventually()`.

When that happens I'd log the errors whenever `err != nil`.

Reply:
No, I'm an against 3. But we should be careful what types of errors we treat as such, e.g. if creating a request fails or issuing an HTTP call fails, it might be a spurious failure which shouldn't be taken into account for our test's results. We can just retry i.e. return false and rerun the predicate logic.

To be honest, it would be better if `require` and `assert` had a permanent error type (like [`backoff` package has](https://pkg.go.dev/github.com/cenkalti/backoff/v4#Permanent)) which would indicate that the retries can be stopped and we should fail the test.